cumulusci/tasks/bulkdata/mapping_parser.py

```diff
@@ -187,8 +187,12 @@ def get_relative_date_context(self, fields: List[str], org_config: OrgConfig):
 
     @validator("batch_size")
     @classmethod
-    def validate_batch_size(cls, v):
-        assert v <= 200 and v > 0
+    def validate_batch_size(cls, v, values):
+        if values["api"] == DataApi.REST:
+            assert 0 < v <= 200, "Max 200 batch_size for REST loads"
+        else:
+            assert values["api"] in (DataApi.SMART, DataApi.BULK)
```

Comment:
I think we should disallow this parameter entirely for `DataApi.SMART`, since the user can't set a default that would be valid for both candidate APIs.

Aren't we breaking backwards compatibility with mapping files here, since the validator will fail if there's no explicit setting?

Reply:
It won't be called if there's no explicit setting. In neither the old code nor the new is it allowed to explicitly pass None.

Reply:
"It won't be called if there's no explicit setting." appears to be true, but I tested changing one of the example mappings to specify `batch_size: null` and the tests still passed. I think that's fine though; there's not much reason to do that but it uses the default batch size if you do.

Reply:
Strange...I thought I saw an exception when I tried that.

Reply:
Anyway, what about @davidmreed's other comment: "I think we should disallow this parameter entirely for DataApi.SMART, since the user can't set a default that would be valid and sensical for both candidate APIs."

Reply:
I started implementing it but I just realized that this change WOULD be backwards compatible, because SMART is the default, and legacy datasets might have set batch_size 150 without changing the API.

I would argue that tomorrow's behaviour of doing both auto-selected REST and auto-selected BULK with the value 150 is a lot more sensical than yesterday's behaviour of completely ignoring the value for BULK.

Reply:
I could add a warning for that case however.

Reply:
Setting a batch size of say 150-200 for the Bulk API can have unpleasant consequences, though - loading a couple of million records will kill the Bulk API batch limit for the next 24 hours.

What if we separated the parameters, since they mean different things? Let `batch_size` control the REST API only, and add `bulk_api_batch_size` for the Bulk API?

Reply:
Wouldn't a warning be simpler for the user than two mutually exclusive integer options?

Reply:
I agree with @davidmreed that since there's potential for both REST and Bulk APIs to be used in the same data load and  the batch size doesn't have the same implications for both of them, it should be a separate setting

Reply:
@davisagli We had a Slack conversation and the heart of it was that it never makes sense to have a REST API batch_mode setting of let's say 150 and a bulk API setting of let's say 500 because the transaction limits that forced you to 150 or the REST API would also force you to 150 or the bulk API.

But in such a case you really shouldn't be using the Bulk API at all. Which implies that you shouldn't be using the smart API setting. Which is why we have a warning.